src/cmd/compile/internal/syntax/walk.go

```diff
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// This file implements syntax tree walking.
+// This file implements syntax tree walking and changing.
```

Comment:
Maybe it's better to not define walk and change as a part of walk file(?).

They defined it to walk over AST, not to change it.

Reply:
Their implementations are almost the same, and I couldn't find a new concept to create a new file. I think we should have them together in one file for now.

Reply:
Ok, so let's consider it as a TODO for the future.